### PR TITLE
Add JAX backend + reduced-spectrum refactor and benchmarks

### PR DESCRIPTION
**Pull request by steven-murray**
## Summary
- add a first-class powerbox.jax API with JAX FFT wrappers and backend wiring
- migrate generation paths to reduced-spectrum rFFT/irFFT workflows for NumPy and JAX
- add/adjust regression coverage for tuple geometry, reduced-spectrum hermitianity, non-cubic power recovery, and JAX lognormal identities
- update backend benchmark tooling/docs and regenerate plots/results
- tune FFTW benchmarking defaults (single-thread in benchmark script) and enable interface-cache reuse in FFTW backend

## Validation
- uv run pytest
- uv run pytest -W error
- uv run pytest tests/test_fft.py
- uv run pytest tests/test_jax.py
- uv run python docs/demos/benchmark_backends.py
- notebook demos via papermill (cosmological_fields, dft, getting_started)

## Notes
- PR targets v1 as requested.

## Summary by Sourcery

Introduce a JAX-backed PowerBox API and refactor Fourier workflows to use reduced-spectrum real FFTs across NumPy, FFTW, and JAX backends.

New Features:
- Add a JAX subpackage providing PowerBox and LogNormalPowerBox generators, FFT wrappers, and power-spectrum tooling built on JAX FFTs.
- Introduce a JAX-based backend benchmark script and documentation pages comparing NumPy, FFTW, and JAX performance.

Bug Fixes:
- Ensure Hermitian symmetry and real-valued real-space fields by centralizing Hermitianization logic and using reduced rFFT/irFFT layouts for spectrum generation.
- Fix handling of non-cubic and mixed odd/even geometries in power recovery and delta_x reality tests.

Enhancements:
- Refactor core PowerBox geometry and FFT paths to operate on tuple-based per-axis shapes and reduced half-spectra, simplifying k-space grid and correlation computations.
- Extend the FFT abstraction layer with real FFT support, JAX and FFTW backends that expose rFFT operations, backend-specific xp namespaces, and improved phase-handling.
- Relax and centralize power-recovery assertions into reusable helpers for more robust regression coverage across backends.

Build:
- Declare JAX as an optional extra and test dependency for JAX-enabled builds and CI.

Documentation:
- Document the JAX API surface and add demos plus backend benchmark plots to the documentation set.

Tests:
- Add extensive NumPy and JAX tests covering reduced-spectrum Hermitianity, real-valued delta_x for various geometries, JAX lognormal identities, and backend cache behavior.
- Update discrete sampling and power-spectrum tests to work with tuple geometry and reduced-spectrum k-vectors.

### CODE REVIEW

File: tests/test_fft.py [hunk 1]
```diff
@@ -133,6 +133,19 @@ def test_mixed_2d_fb(g2d, a, b, ainv, binv, backend) -> None:
...
+@pytest.mark.skipif(not HAVE_FFTW, reason="pyFFTW not installed")
+def test_fftw_backend_enables_interface_cache() -> None:
+    """The FFTW backend should enable the pyFFTW interface cache for plan reuse."""
+    import pyfftw
+
+    pyfftw.interfaces.cache.disable()
+    assert not pyfftw.interfaces.cache.is_enabled()
+
+    FFTW(nthreads=1)
+
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply tests/test_fft.py:151]: Done.

File: tests/test_jax.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+def test_jax_get_power_partial_average_retains_unbinned_axes() -> None:
+    pb = jpb.PowerBox(
+        (18, 20, 22),
+        dim=3,
+        pk=lambda k: (1 + k) ** -2.0,
+        boxlength=(4.0, 5.0, 6.0),
+        key=jax.random.key(2),
+        b=1,
+    )
+
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply tests/test_jax.py:208]: Done

File: src/powerbox/jax/tools.py [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+
+    average_field = _field_average(indx, field, weights, sumweights)
+    variance = None
+    if get_variance:
+        variance = _field_variance(indx, field, average_field, weights, sumweights)
+
+    return average_field, bins_out, variance, sumweights
+
+
+def angular_average_nd(
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply src/powerbox/jax/tools.py:373]: Deferred via #143

File: src/powerbox/jax/powerbox.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    if np.isscalar(obj) or np.ndim(obj) == 0:
+        return (convert(obj),) * dim
+
+    if len(obj) != dim:
+        raise ValueError(f"{name} must be a scalar or have length {dim}.")
+
+    return tuple(convert(value) for value in obj)
+
+
+def _sample_gaussian_hermitian_fft(axis_lengths: tuple[int, ...], key: jax.Array) -> jax.Array:
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply src/powerbox/jax/powerbox.py:18]: Deferred via #142
